### ISSUE TITLE
Use ssh keys in stead of password

### ISSUE THREAD

**Issue by NillsF** [richardspitz/imagefactory#1]
As your VM is temporary in nature, why not use SSH keys in stead of passwords?

https://docs.microsoft.com/en-us/powershell/module/azurerm.compute/add-azurermvmsshpublickey?view=azurermps-6.13.0 should help you out with it.

**Comment by richardspitz** [comment]
Good point. However, I wanted the solution to be simple to use for non-Linux users who aren't familiar with ssh keys. We could use ssh keys internally however that would take away the option of users signing in to investigate in case of failures.

Thoughts?

**Comment by NillsF** [comment]
As the VMs are stateless, I don't see a need for users to have to login. I understand the need to login during troubleshooting - but during regular runs you don't have to login I assume.

**Comment by richardspitz** [comment]
Quick question. Since we're doing an ARM deployment of the temporary VM, this would require users to specify a public key as a runbook variable which can then be passed to the template as a parameter. This would mean users would need to be familiar with ssh keys, creating an external dependency.
